app/grandchallenge/components/models.py

```diff
@@ -264,6 +305,10 @@ class ComponentInterfaceValue(models.Model):
         to=Image, null=True, blank=True, on_delete=models.CASCADE
     )
 
+    @property
+    def has_value(self):
+        return self.value is not None or self.image or self.file
```

Comment:
Could you add a docstring re: the intention of this propetry? Sometimes a value of `None` _is_ valid.

Reply:
We currently use this to check if all civs have a value/file/image before actually running the algorithm. In the form all inputs are required, so there should not be any None values I'd say. In what situation would a None value be valid?

Reply:
There was some discussion about it for (multi) choice, but yes, I _think_ that all of the values are non-nullable.

Reply:
Wouldn't it make more sense for the value to be `[]` in case of a multiple choice with no selected values?

Reply:
Yes, discussion here https://github.com/DIAGNijmegen/rse-roadmap/issues/9#issuecomment-799579824